src/plugins/validation/2and3/semantic-validators/responses.js

```diff
@@ -39,7 +39,11 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
                     ) {
                       const hasInlineSchema = !mediaType.schema[schemaType][i]
                         .$ref;
-                      if (hasInlineSchema) {
+                      const arrItemsIsRef =
```

Comment:
I hate to add this complication, but while reviewing the testcase below, I was reminded of the fact that we should also "exempt" scenarios where we have an array schema whose items type is a primitive.   
This is because the generator "flattens" away refs to an array schema with a primitive-type "items" the same that it does it for refs to an array schema with a ref-type "items".

Reply:
We can handle that! I asked it on another comment as well, but I'll duplicate the question. Should it only be an exemption for string, number, integer, and boolean?